src/InfoBoxes/InfoBoxLayout.cpp

```diff
@@ -22,12 +22,11 @@ Copyright_License {
 */
 
 #include "InfoBoxes/InfoBoxLayout.hpp"
+#include "Interface.hpp"
```

Comment:
Don't include this header. `InfoBoxLayout.cpp` is supposed to be a low-level library without too many dependencies, and `Interface.hpp` is maybe the heaviest dependency of all.
Instead, pass this new setting as another parameter.
For the same reason, the `InfoBoxSettings::Geometry` is passed as parameter instead of loading it from the global variables.

Reply:
OK, I understand. I thought CommonInterface::GetUISettings().info_boxes.ib_to_screen_ratio would compile to not much more than a memory read.

I need to see how to fix it. May take some time.

Reply:
> I thought CommonInterface::GetUISettings().info_boxes.ib_to_screen_ratio would compile to not much more than a memory read.

Your thought is correct. But this is not about runtime overhead, it's about API bloat (on the source code level, not at runtime). Adding a runtime dependency on `Interface.hpp` means we cannot write a unit test for this code, for example.

Reply:
Do you want me to add a 3rd parameter to InfoBoxLayout::Calculate() ??

Reply:
Or just add a variable to the struct InfoBoxLayout::Layout ??
(But then I have to include  InfoBoxLayout into Profile/InfoBoxConfig.cpp which might be wash.)